src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+              match = allDna.get(stock.getId());
+              match.addRunLane(run, lane);
+              classifyIfActiveThisMonth(run, stock, match, dnaThisMonth);
+              allDna.put(stock.getId(), match);
+            } else {
+              match = allRna.get(stock.getId());
+              match.addRunLane(run, lane);
+              classifyIfActiveThisMonth(run, stock, match, rnaThisMonth);
+              allRna.put(stock.getId(), match);
+            }
+          }
+        }
+      }
+    }
+    dnaRecentList = listifyMap(dnaThisMonth);
+    rnaRecentList = listifyMap(rnaThisMonth);
+    dnaAllList = listifyMap(allDna);
+    rnaAllList = listifyMap(allRna);
+    // reverse the "all" lists, as we want to display most recent items at the top
+    Collections.reverse(dnaAllList);
+    Collections.reverse(rnaAllList);
+  }
+
+  private void addReportItem(ReportItem item, Map<String, ReportItem> recent, Map<String, ReportItem> all) {
+    Set<SampleDto> recentAliquots = item.getAliquots().stream()
+        .filter(aliq -> isCreatedBetween(aliq, start, end))
+        .collect(Collectors.toSet());
+    if (!recentAliquots.isEmpty() || isCreatedBetween(item.getStock(), start, end)) {
+      recent.put(item.getStock().getId(), item);
+    }
+    all.put(item.getStock().getId(), item);
+  }
+
+  private static final ReportItem addLibToReportItem(SampleDto stock, SampleDto lib, Map<String, ReportItem> all,
+      Map<String, SampleDto> allSamples) {
+    String groupId = getAttribute(ATTR_GROUP_ID, lib);
+    if (groupId == null) groupId = getUpstreamAttribute(ATTR_GROUP_ID, lib, allSamples);
+    ReportItem match = all.get(stock.getId());
+    if (match == null) {
+      // SIGH. some test projects use aliquots from different projects
+      SampleDto identity = getParent(stock, SAMPLE_CATEGORY_IDENTITY, allSamples);
+      String externalName = getAttribute(ATTR_EXTERNAL_NAME, identity);
+      ReportItem deNovo = new ReportItem(stock, externalName);
+      match = deNovo;
+    }
+    match.addLibrary(lib, groupId);
+    return match;
+  }
+
+  private final void classifyIfActiveThisMonth(RunDto run, SampleDto stock, ReportItem match, Map<String, ReportItem> thisMonth) {
+    if ((RUN_COMPLETED.equals(run.getState())
+        && run.getCompletionDate().compareTo(start) > 0
+        && run.getCompletionDate().compareTo(end) < 0)
+        ||
+        (RUN_RUNNING.equals(run.getState()) &&
+            (run.getStartDate().compareTo(start) > 0
+                && run.getStartDate().compareTo(end) < 0))) {
+      thisMonth.put(stock.getId(), match);
+    }
+  }
+
+  private static final List<Map.Entry<String, ReportItem>> listifyMap(Map<String, ReportItem> map) {
+    // need to convert it to a list, because getRow() calls item by index
+    ArrayList<Map.Entry<String, ReportItem>> listified = new ArrayList<>(map.entrySet());
+    Collections.sort(listified, new Comparator<Map.Entry<String, ReportItem>>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)